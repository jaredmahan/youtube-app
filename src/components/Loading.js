import React from "react";

export const Loading = ({ loadingText, loading, children }) =>
  loading ? <div>{loadingText}</div> : children;

export default Loading;
