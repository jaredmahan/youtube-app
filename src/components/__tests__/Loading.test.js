import React from "react";
import { Loading } from "../Loading";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<Loading>", () => {
  it("renders correctly when loading", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Loading loading={true} loadingText="Loading...">Test</Loading>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it("renders correctly when not loading", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Loading loading={false} loadingText="Loading...">Test</Loading>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
