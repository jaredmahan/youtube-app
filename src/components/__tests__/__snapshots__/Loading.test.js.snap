// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading> renders correctly when loading 1`] = `
<div>
  Loading...
</div>
`;

exports[`<Loading> renders correctly when not loading 1`] = `"Test"`;
